php/helpers.php

```diff
@@ -25,13 +25,27 @@ function block_field( $name, $echo = true ) {
 	 */
 	global $block_lab_attributes, $block_lab_config;
 
+	$default_fields = array( 'className' );
+
 	if ( ! isset( $block_lab_attributes ) || ! is_array( $block_lab_attributes ) ) {
 		return null;
 	}
 
-	$default_fields = array( 'className' );
+	$is_name_allowed = isset( $block_lab_config->fields[ $name ] ) || in_array( $name, $default_fields, true );
+
+	/**
+	 * Filters whether the $name argument to this function is allowed.
+	 *
+	 * By default, this only allows a $name that is in the block fields or default fields, like 'className'.
+	 * Returning true to this filter can prevent exiting from block_field()
+	 * when looking for another value in the block attributes.
+	 *
+	 * @param bool   $is_name_allowed Whether the name is allowed.
+	 * @param string $name The name of value to get.
+	 */
+	$is_block_field_name_allowed = apply_filters( 'is_block_field_name_allowed', $is_name_allowed, $name );
```

Comment:
Why are we filtering the `is_name_allowed` result, instead of filtering the `default_fields` array?

Reply:
Ah, that's a good idea. I didn't think of that, as the `$default_fields` array didn't exist when I originally wrote this filter in another branch, about a month ago.

Reply:
Ha! I only picked up on this because I thought you pulled me up on this a couple weeks back. :P

Reply:
Maybe just do whatever we are already doing elsewhere in the plugin.

Reply:
You make a good point, I think it's simpler to filter the `$default_fields`.

Reply:
And thanks to the simplicity, there's less risk of regression.